agents/pfeiffer_tpg366/pfeiffer_tpg366_agent.py

```diff
@@ -74,10 +93,15 @@ def read_pressure_all(self):
         self.comm.send(ENQ.encode())
         read_str = self.comm.recv(BUFF_SIZE).decode()
         pressure_str = read_str.split('\r')[0]
-        #gauge_states = pressure_str.split(',')[::2]
-        #gauge_states = np.array(gauge_states, dtype=int)
+        gauge_states = pressure_str.split(',')[::2]
+        gauge_states = np.array(gauge_states, dtype=int)
         pressures = pressure_str.split(',')[1::2]
         pressures = [float(p) for p in pressures]
+        if any(state != 0 for state in gauge_states):
+            index = np.where(gauge_states != 0)
+            for j in range(len(index[0])):
+                channel = np.int(index[0][j])
+                pressures[channel] = 0.
```

Comment:
A few things here:
- What's the expected pressure value when a sensor is off that you're setting to `0.` now?
- Looking at the manual for comms with the Pfeiffer device, there are more states than just `0` for OK, and non-zero for "Sensor switched off". State 5 seems to indicate no gauge and outputs `2.0E-2` for pressure. Are other states of interest? And do they return values other than `2.0E-2`? They could provide more information about a sensor that right now would make it appear to just be "off", when it could mean under or over range, error, no gauge, or ID error.
- Rather than doing `for j in range(len(index[0])):` you should use the [enumerate()](https://docs.python.org/3/library/functions.html#enumerate) built-in function. But even then, you don't really need the index here, and could get away with:
```
for j in index[0]:
    pressures[j] = 0.
```
- `np.int` [is deprecated](https://numpy.org/devdocs/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated), use `int`, though I believe that `np.where` should give you only integers already, since it's the indices where `gauge_states != 0`, so I excluded the casting in the above example.

Reply:
To answer a few of my questions:
1. When sensors are off they read zero. Here's example output from a 'PRX' command with our DR's TPG366, p1 is off, the rest are on:
```
b'4,+0.0000E+00,0,+7.8800E+00,0,+7.0160E+01,0,+1.7824E+01,0,+2.2000E+01,0,+1.9500E+01\r\n'
```
2. Other states probably are useful, but are currently not checked. Will implement separately so we can consider this resolved for now. The rest of the changes in this comment were implemented.